### PR TITLE
[ACA-4423] VIEWER - when closing the upload dialog the filename changes to "Unknown"

### PR DESCRIPTION
**Pull request by ursedaniel**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

Forgot in the original PR to remove the blob line and after update of the node the viewer acts as a blob instead of the image from ACS.

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**: https://alfresco.atlassian.net/browse/ACA-4423

### CODE REVIEW

**Review by montgolfiere — APPROVED** [review]

**Review by PunchedChimera — APPROVED** [review]

File: lib/core/viewer/components/viewer.component.spec.ts [hunk 1]
```diff
@@ -962,25 +965,27 @@ describe('ViewerComponent', () => {
...
+                spyOn(uploadService, 'uploadFilesInTheQueue').and.callFake(() => {});
                 component.readOnly = false;
                 component.nodeEntry = new NodeEntry({ entry: { name: 'fakeImage.png', id: '12', content: { mimeType: 'txt' } } });
                 const data = atob('iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==');
                 const fakeBlob = new Blob([data], { type: 'image/png' });
                 component.onSubmitFile(fakeBlob);
                 fixture.detectChanges();
 
-                expect(component.blobFile).toEqual(fakeBlob);
+                expect(uploadService.uploadFilesInTheQueue).toHaveBeenCalled();
```
Comment (mauriziovitale) [inline lib/core/viewer/components/viewer.component.spec.ts:976]: why are we changing that? before we were checking the file and now only that a method is called which does not make any sense.

File: lib/core/viewer/components/viewer.component.spec.ts [hunk 2]
```diff
@@ -962,25 +965,27 @@ describe('ViewerComponent', () => {
...
+                spyOn(uploadService, 'uploadFilesInTheQueue').and.callFake(() => {});
                 component.readOnly = true;
                 component.nodeEntry = new NodeEntry({ entry: { name: 'fakeImage.png', id: '12', content: { mimeType: 'txt' } } });
                 const data = atob('iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==');
                 const fakeBlob = new Blob([data], { type: 'image/png' });
                 component.onSubmitFile(fakeBlob);
                 fixture.detectChanges();
 
-                expect(component.blobFile).toEqual(undefined);
+                expect(uploadService.uploadFilesInTheQueue).not.toHaveBeenCalled();
```
Comment (mauriziovitale) [inline lib/core/viewer/components/viewer.component.spec.ts:988]: same as before

**Review by mauriziovitale — CHANGES_REQUESTED** [review]
comments

**Review by eromano — CHANGES_REQUESTED** [review]
can we avoid reloading the image viewer when we save? I don't think is necessary. This is actually also creating some flickering.

File: lib/core/viewer/components/viewer.component.spec.ts (hunk 1 above)
Comment (ursedaniel) [reply lib/core/viewer/components/viewer.component.spec.ts:976]: because that line with blob was added by mistake, we don't need that

Comment (ursedaniel) [reply lib/core/viewer/components/viewer.component.spec.ts:976]: i'll also add an expect to check if the upload was called with the right BLOB.

**Comment by ursedaniel** [comment]
as investigated with @eromano for the moment ACA always refreshes the viewer based on the subscription for uploadService so we will investigate in the next release the possibility to stop always refreshing the viewer after upload

**Review by DenysVuika — APPROVED** [review]

**Review by eromano — APPROVED** [review]

**Review by mauriziovitale — APPROVED** [review]

**Review by mauriziovitale — APPROVED** [review]

**Pull request merged**